### PR TITLE
Adding Microsoft Azure Logs sample and Windows Perfmon option

### PR DESCRIPTION
**Pull request by harrykimpel**

### CODE REVIEW

File: examples/windows/windows-perfmon-proc-owner.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+integrations:
+  - name: nri-flex # We're telling the Infra agent to run Flex!
+    config: # Flex configuration starts here!
+      name: msPerfProcOwner
+      apis:
+        # Event Requests
+        - event_type: msPerfProcOwner # Name of the event in New Relic
+          commands:
+            - run: "C:/Program Files/New Relic/newrelic-infra/integrations.d/windows-perfmon-proc-owner.bat"
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.yml:9]: please remove the batch file (`.bat`) dependency here and add the `shell` attribute to the config so you can execute the PowerShell script natively. Example [HERE](https://github.com/newrelic/nri-flex/blob/master/examples/windows/windows-ping-example.yml)

File: examples/windows/windows-perfmon-proc-owner.bat [hunk 2]
```diff
@@ -0,0 +1,2 @@
+@echo off
+Powershell -File "C:\Program Files\New Relic\newrelic-infra\integrations.d\windows-perfmon-proc-owner.ps1"
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.bat:2]: please remove the batch file (`.bat`) dependency here and add the `shell` attribute to the config so you can execute the PowerShell script natively. Example [HERE](https://github.com/newrelic/nri-flex/blob/master/examples/windows/windows-ping-example.yml)

File: examples/windows/windows-perfmon-proc-owner.ps1 [hunk 3]
```diff
@@ -0,0 +1,44 @@
+$computername = "DESKTOP-26GUTUU"
+$colProcs = Get-wmiobject win32_process -computername $computername  -Credential $credential | select *, @{Name = "Owner"; Expression = { ($_.GetOwner()).User } }
+$colPerfs = Get-wmiobject win32_perfformatteddata_perfproc_process -computername $computername  -Credential $credential
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.ps1:3]: please update these to `Get-CimInstance` to align with future direction of MSFT and reduce load of query against system

File: examples/windows/windows-perfmon-proc-owner.ps1 [hunk 4]
```diff
@@ -0,0 +1,44 @@
+$computername = "DESKTOP-26GUTUU"
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.ps1:1]: should this be redacted?

File: examples/windows/windows-perfmon-proc-owner.ps1 [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+$colProcs = Get-wmiobject win32_process -computername $computername  -Credential $credential | select *, @{Name = "Owner"; Expression = { ($_.GetOwner()).User } }
+$colPerfs = Get-wmiobject win32_perfformatteddata_perfproc_process -computername $computername  -Credential $credential
+$colTasklist = @()
+ 
+foreach ($perf in $colPerfs) {
+    if ($perf.Name.StartsWith("w3wp")) {
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.ps1:7]: filtering should be pushed into the `Get-CimInstance` currently `get-wmiobject` to reduce query load

**Review by thezackm — CHANGES_REQUESTED** [review]
Azure sample looks solid; the Perfmon sample needs some updates

File: examples/windows/windows-perfmon-proc-owner.yml (hunk 1 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.yml]: Removed dependency on bat file

File: examples/windows/windows-perfmon-proc-owner.ps1 (hunk 4 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.ps1]: updated script

File: examples/windows/windows-perfmon-proc-owner.bat (hunk 2 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.bat]: done

File: examples/windows/windows-perfmon-proc-owner.ps1 (hunk 3 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.ps1]: as for win32_perfformatteddata_perfproc_process I can replace this with Get-CimInstance, the first one does not provide the owner when changing from Get-WmiObject to Get-CimInstance

File: examples/windows/windows-perfmon-proc-owner.ps1 (hunk 5 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.ps1]: fixed

**Comment by harrykimpel** [comment]
@thezackm could you pls. have a look again and review my changes?

File: examples/microsoft-azure-monitor-logs.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  - name: nri-flex # We're telling the Infra agent to run Flex!
+    config: # Flex configuration starts here!
+      name: msAzureMonitorLogs
+      apis:
+        # Event Requests
+        - event_type: msAzureMonitorStatus # Name of the event in New Relic
+          commands:
+            #- run: /etc/newrelic-infra/integrations.d/exec.sh
+            - run: dotnet /etc/newrelic-infra/integrations.d/ms_azure_monitor_event_hubs_receiver.dll <NEWRELIC_LICENSE_KEY>
+              timeout: 59000
```
Comment (thezackm) [inline examples/microsoft-azure-monitor-logs.yml:11]: the default 30s interval is still applied at the Integrations SDK level here; is a 59s timeout going to cause conflicts?

File: examples/windows/windows-perfmon-proc-owner.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  - name: nri-flex # We're telling the Infra agent to run Flex!
+    interval: 60s
+    timeout: 45s
+    config: # Flex configuration starts here!
+      name: msPerfProcOwner
+      apis:
+        # Event Requests
+        - event_type: msPerfProcOwner # Name of the event in New Relic
+          commands:
+            - run: "& \"C:/Program Files/New Relic/newrelic-infra/integrations.d/windows-perfmon-proc-owner\""
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.yml:11]: missing the file extension on this command

File: examples/windows/windows-perfmon-proc-owner.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+integrations:
+  - name: nri-flex # We're telling the Infra agent to run Flex!
+    interval: 60s
+    timeout: 45s
```
Comment (thezackm) [inline examples/windows/windows-perfmon-proc-owner.yml:4]: curious: is it necessary to extend this past the default 30s if the Flex timeout below is also 30s?

File: examples/microsoft-azure-monitor-logs.yml (hunk 6 above)
Comment (harrykimpel) [reply examples/microsoft-azure-monitor-logs.yml:11]: I don't think so

File: examples/windows/windows-perfmon-proc-owner.yml (hunk 8 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.yml:4]: probably not

File: examples/windows/windows-perfmon-proc-owner.yml (hunk 7 above)
Comment (harrykimpel) [reply examples/windows/windows-perfmon-proc-owner.yml:11]: works fine like this ... referred to this https://github.com/newrelic/nri-flex/blob/master/examples/windows/windows-ping-example.yml

**Review by thezackm — APPROVED** [review]
let's GTM! 🚀

**Comment by thezackm** [comment]
@Kav91 - can you merge this one; the CLA verification is still broken and requires admin rights to override

**Pull request merged**